lib/llm/conversation.rb

```diff
@@ -2,23 +2,40 @@
 
 module LLM
   class Conversation
```

Comment:
@antaz Maybe we should have two types of Conversation: synchronous and async / (lazy). This class could become `LazyConversation`. And you could enter with `Provider#lazy_chat` or something like that. Wdyt?